src/mlpack/core/metrics/bleu_impl.hpp

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file core/metrics/bleu_impl.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Implementation of BLEU class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_BLEU_IMPL_HPP
+#define MLPACK_CORE_METRICS_BLEU_IMPL_HPP
+
+// In case it hasn't been included.
+#include "bleu.hpp"
+
+namespace mlpack {
+namespace metric {
+
+template <typename ElemType, typename PrecisionType>
+BLEU<ElemType, PrecisionType>::BLEU(const size_t maxOrder) :
+  maxOrder(maxOrder),
+  translationLength(0),
+  referenceLength(0)
+{
+  // Nothing to do here.
+}
+
+template <typename ElemType, typename PrecisionType>
+template <typename WordVector>
+std::map<WordVector, size_t> BLEU<ElemType, PrecisionType>::GetNGrams(
```

Comment:
>It could be interesting to compare the performance and take into account all pros and cons.

I will look into it but I am not sure how much time it can take :\

Reply:
> I will look into it but I am not sure how much time it can take :\

Then leave as is. It's not a high priority task.